src/libopensc/reader-pcsc.c

```diff
@@ -1147,13 +1149,12 @@ static void detect_reader_features(sc_reader_t *reader, SCARDHANDLE card_handle)
 
 	sc_log(ctx, "Requesting reader features ... ");
 
-	if (gpriv->SCardControl == NULL)
-		return;
-
-	rv = gpriv->SCardControl(card_handle, CM_IOCTL_GET_FEATURE_REQUEST, NULL, 0, buf, sizeof(buf), &rcount);
-	if (rv != SCARD_S_SUCCESS) {
-		PCSC_TRACE(reader, "SCardControl failed", rv);
-		return;
+	if (gpriv->SCardControl != NULL) {
+		rv = gpriv->SCardControl(card_handle, CM_IOCTL_GET_FEATURE_REQUEST, NULL, 0, buf, sizeof(buf), &rcount);
+		if (rv != SCARD_S_SUCCESS) {
+			PCSC_TRACE(reader, "SCardControl failed", rv);
+			return;
+		}
 	}
 
 	if ((rcount % sizeof(PCSC_TLV_STRUCTURE)) != 0
```

Comment:
This makes an unitializer read of `rcount` if there is no SCardControl. I think initialization should initialize it to 0, which should do the job.

Reply:
AFAIK, that is used as output parameter only. I'll add the initialization anyway, it doesn't hurt.